src/loader.js

```diff
@@ -0,0 +1,91 @@
+let assume = require('assume');
+let debug = require('debug')('taskcluster:loader');
+let Promise = require('promise');
+
+function loader(componentDirectory, requiredVirtualComponents) {
+  /**
+   * If present, this is a list of component names that a loader
+   * must define.  These are likely things like configuration values
+   */
+  if (requiredVirtualComponents) {
+    assume(requiredVirtualComponents).is.array();
+  } else {
+    requiredVirtualComponents = [];
+  }
+
```

Comment:
One of the benefits here is that you can statically validate that dependency relations in `componentDirectory` is a DAG.
I know it's boring to code: http://stackoverflow.com/questions/4168/graph-serialization/4577#4577
But we have npm: https://www.npmjs.com/package/topo-sort
It'll throw an error if it's not a DAG :)